### DISCUSSION TITLE
` v50.0-beta1 Pre-release` feedback

### DISCUSSION THREAD

**Discussion by joshinils** [streetcomplete/StreetComplete#4670] [Feedback]
For separate exclusive cycleways in germany which are not designated via sign i would like another colour than black. those are cycleways, ie. `highway=cycleway` + `bicycle=yes` + `traffic_sign=none`.

for those situations i can not distinguish wether the street is correctly tagged as having a separate bicycle path or if that is an exclusive sidewalk for pedestrians only, or if there is a separate bicycle path but _also_ a separate cycleway tagged on the road-way (i.e. duplicate tagging of the same cycleway).

there is no way to tag a way as allowed but not designated (which happens a lot in germany especially i feel).

the sign-combination `DE:239` + `DE:1022-10`  <img src="https://upload.wikimedia.org/wikipedia/commons/5/5a/Zeichen_239_-_Sonderweg_Fu%C3%9Fg%C3%A4nger%2C_StVO_1992.svg" height=25><img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Zusatzzeichen_1022-10_-_Radfahrer_frei%2C_StVO_1992.svg" height=20> means the path is designated for pedestrians `highway=footway` + `footway=sidewalk` (+ `foot=designated`) , but cycling is allowed `bicycle=yes` but only while taking care of pedestrians (have the right of way) and only going at walking pace `max_speed:bicycle=walk`. 
those footpaths are also black, which seems right, but also means no distinction between other non-cycling paths.

**Comment by joshinils** [comment]
where there is a `oneway=yes` tag but no oneway-sign, so for streets with a median it makes no sense to ask for the left side (it would probably be tagged on the opposide side of the dual carriageway). but OSM knows no distinction between "real" oneway roads with `traffic_sign=DE:220` <img src="https://upload.wikimedia.org/wikipedia/commons/9/94/Zeichen_220-20_-_Einbahnstra%C3%9Fe%2C_rechtsweisend%2C_StVO_2017.svg" height=20> (where you may park on the left hand side in germany) and "fake" oneway roads only tagged as such for routing and logical reasons.

the only way around this i see is to use the tag [`dual_carriageway=yes`](https://wiki.openstreetmap.org/wiki/Tag:dual_carriageway%3Dyes) if it exists. but it probably is best to explicitly tag all those as not having a cycleway on the left hand side. same situation with the parking overlay.

**Comment by joshinils** [comment]
btw. here are changesets made with `StreetComplete:quest_type=CyclewayOverlay` 
https://osmcha.org/?filters=%7B%22metadata%22%3A%5B%7B%22label%22%3A%22StreetComplete%3Aquest_type%3DCyclewayOverlay%22%2C%22value%22%3A%22StreetComplete%3Aquest_type%3DCyclewayOverlay%22%7D%5D%7D

**Comment by westnordost** [comment]
> For separate exclusive cycleways in germany which are not designated via sign i would like another colour than black. those are cycleways, ie. highway=cycleway + bicycle=yes + traffic_sign=none.

`traffic_sign=none` doesn't seem to be that well established. It's not even documented at all. Also, filtering by `traffic_sign=none` wouldn't work per se, because there may always be other (unrelated) traffic signs.
Anyway, that an unsigned cycleway is not classified as "something like a footway" is fine, will change that.

> there is no way to tag a way as allowed but not designated (which happens a lot in germany especially i feel).
> the sign-combination DE:239 + DE:1022-10 means the path is designated for pedestrians highway=footway + footway=sidewalk (+ foot=designated) , but cycling is allowed bicycle=yes but only while taking care of pedestrians (have the right of way) and only going at walking pace max_speed:bicycle=walk.
those footpaths are also black, which seems right, but also means no distinction between other non-cycling paths.

Yes, no distinction is made between footway with bicycles allowed and footway with no bicycles allowed. 
See https://github.com/streetcomplete/StreetComplete/issues/4669 for reaons.

**Reply by westnordost** [reply]
> Anyway, that an unsigned cycleway is not classified as "something like a footway" is fine, will change that.

By the way, as `highway=footway` per wiki documentation implies `foot=designated`, `foot=designated` cannot be interpreted as that "there is always a sign". This is true especially outside of Germany's / Europe's "sign jungle" but even in Germany, you will find plenty of ways that were clearly purpose-built for pedestrians and are **not** signed explicitly with the blue lollipop. For example, just about any footway that branches off from a sidewalk towards apartment complexes. Still, these ways are de-facto very often tagged as `highway=footway` and not something like `highway=path`. Simply because they have been intended (i.e. purpose-built) for pedestrians, which is the other interpretation of `designated`.

Conversely, the same is true for cycleways.

In conclusion, one can say that
1. the wiki contradicts itself in a few places
2. what exactly `designated` means is subject to different interpretations, in particular
    1. , some say "it must be signed" (because this is what the word means). This interpretation doesn't really work outside of the German/European "sign jungle", subsequently, most footways even in the US would need to be tagged always as `highway=path` in the absence of such designation. However, this is not the case. Additionally, presets in iD and JOSM call e.g. a `highway=footway` just a "footway", not a "designated footway"
    2. some say it means that "it was made with that mode of transport as primary user in mind", i.e. pedestrians on footways, cyclists on cycleways etc..

**Reply by matkoniecz** [reply]
I would say that `foot=designated` is much weaker and requiring less signage than `bicycle=designated` (at least in Poland)

**Reply by westnordost** [reply]
Hmm, yeah, I spent the last 20 minutes reading through the wiki again and I wonder if `highway=footway` should just be imply `foot=yes` / used interchangeably with `foot=designated` instead.

**Reply by westnordost** [reply]
On the other hand, 

```
highway=path
foot=designated
bicycle=yes
```
or 
```
highway=footway
bicycle=yes
```
*is* documented to be different from
```
highway=path
bicycle=yes
```

Latter is "just a path", former is a footway on which cyclists are (merely) allowed.

**Comment by westnordost** [comment]
Okay, I reviewed everything again and came to this, finally. This however means, sorry @joshinils , that `highway=cycleway + bicycle=yes` is parsed as "not designated for cyclists", which actually really is what it is, otherwise it would be `bicycle=designated`.
<img width="649" alt="Screenshot 2022-12-10 at 18 38 09" src="https://user-images.githubusercontent.com/4661658/206868060-0129bc15-0375-4022-9c0d-c9a126623a2f.png">

**Reply by joshinils** [reply]
Yeah, ok. The trouble stems from the different use of `designated` vs `yes` when looking at allowed bicycle usage in Germany specifically.
Simply because there is the option of a cycleway which is optional (you don't have to use it) and one with the blue lollipop meaning you legally have to use it in almost all circumstances (and a centerline accompanying a separately mapped cyclable way should have a `bicycle=use_sidepath` which equals `bicycle=no`).

There is no other way to differentiate between an obligatory cycleway and an optional cycleway via the bicycle-tag other than using designated vs yes.
Another distinction could be the presence of a traffic sign, for example `traffic_sign=DE:240`, but the lack of a blue traffic sign with a white bike on it can only (explicitly) be expressed via `traffic_sign=none`, since a missing tag means nothing.

At least that is my understanding of how and why these values are used differently in Germany.

So in a way _all_ ways where cycling is permitted are logically to be tagged as bicycle=designated, but that would be a great mistake.

I wish there were an option for distinguishing obligatory and optional cycleways.

**Reply by matkoniecz** [reply]
> I wish there were an option for distinguishing obligatory and optional cycleways.

Note that if there is not one you can invent it. Also, as I understand, in some areas they tag traffic sign codes on cycleways to record such distinction.

**Reply by joshinils** [reply]
> > I wish there were an option for distinguishing obligatory and optional cycleways.
> 
> Note that if there is not one you can invent it. 

So far, I am not comfortable starting a Proposal process and discussions.
> Also, as I understand, in some areas they tag traffic sign codes on cycleways to record such distinction.

yeah, we like to do that, eg. `DE:239,1022-10` or `DE:237` but that means one has to tag them, and know how to do it correctly and also connect all the separately mapped cycleways at intersections and tag the centerline with `bicycle=<use_sidepath|optional_sidepath> as applicable and maybe ` cycleway:right=<separate|share_sidewalk>`.
know how to distinguish an optional cycleway from a parking strip on the edge of a kerb, etc.. This sometimes needs more than someone going: "here is this blue sign, let's tag that", especially the optional ones... (which is why i am opposed to them in most cases, since they usually do not meet width, space, buffer to car parking, surface or smoothness requirements)

**Reply by westnordost** [reply]
Hm, the issue with `bicycle=use_sidepath` is,  what about if it differs per side? Ah actually, nvm, there is`bicycle:forward=use_sidepath` etc.

`cycleway=share_sidewalk` is marked as invalid by SC by the way because it is ambiguous (gemeinsamer fuß und radweg oder radfahrer frei?)

**Reply by joshinils** [reply]
> `cycleway=share_sidewalk` is marked as invalid by SC by the way because it is ambiguous (gemeinsamer fuß und radweg oder radfahrer frei?)

good point, I will relay that to [the berlin matrix group](https://matrix.to/#/#osmberlinverkehr:matrix.org)

**Reply by westnordost** [reply]
The "proper" tagging to differentiate the two if it is tagged on the centerline is:

Gemeinsamer Fuß- und Radweg:
```
cycleway:*=track
cycleway:*:segregated=no
```

Fußweg, Fahrradfahrer frei
```
cycleway:*=no
sidewalk:*:bicycle=yes
```

**Comment by joshinils** [comment]
bad example:
https://www.openstreetmap.org/way/479700952
https://www.mapillary.com/app/?lat=52.4481492&lng=13.4387164&z=17&pKey=167627685366360&focus=photo
https://osmlab.github.io/osm-deep-history/#/way/479700952
https://www.openstreetmap.org/changeset/129953845
https://osmcha.org/changesets/129953845/ (osmcha doesn't load for me...)

the footway was previously tagged with
`highway=footway`
`foot=designated`
`bicycle=yes`
`segregated=no` (but since 2017, this was probably in error)
`oneway:bicycle=yes`

and the changeset made changes to 
`bicycle=yes` -> `bicycle=designated` (there is no sign, this is wrong, it is only optional)
`segregated=no` -> `segregated=yes` (this seems ok according to mapillary imagery)

which is wrong, because the cycleway portion is not obligatory, and the [centerline way](https://www.openstreetmap.org/way/600378291) should not be tagged with `bicycle=use_sidepath` but rather `bicycle=optional_sidepath`. It currently is neither.

according to the image of the table on when to display what SC should have shown this piece as:
previously: not designated for cyclists 
now: shared-use path / separated bike and foot path
(my version shows "Not designated for cyclists (regardless of whether cycling is allowed or not)" for a similarly tagged share-use way, i guess this is the same just without the add-on in the parentheses)

I fear this has led to confusion and or misinterpretation and thus mistagging.
The problem I see here is that the shared use path was previously black, and now blue, thus satisfying the user as having corrected a footway into a shared path which allows cycling.

**Reply by westnordost** [reply]
> bicycle=yes -> bicycle=designated (there is no sign, this is wrong, it is only optional)

No, `designated` says nothing about whether people have to use it. That is what `bicycle=use_sidepath`is for. Even in the "it has to be explicitly designated" kind of interpretation of `designated`, it is clear through the red paving that it has been designated for cyclists.

And as **physical** cycleway infrastructure is concerned, this looks to be a (good) separated cycle track. I am not sure why the municipality would not mandate that this cycleway must be used by cyclists. Maybe has to do with the 30 km/h maxspeed, as at least in 30 km/h zones, **no cycle track may be signed as mandatory** (this is by law in Germany) and there is simply no sign that says "hello, here is a cycle track" without also mandating its use. Maybe a white bicycle pictogram on the lane (as that has no legal relevance), but this is not common in Germany.

**Reply by joshinils** [reply]
Exactly that is the point of contention for Germany!
We tag `bicycle=designated` _only_ when there is a round, blue sign with a white bicycle icon on it.
I know the wiki says otherwise.

**Reply by westnordost** [reply]
If you mean with "we" German mappers, I am not so sure about this claim. As you say yourself, the wiki says otherwise and/or is contradictory in a few detailed points such as possibly this one. As there has been no software support to define these different categories so far, it was really up to the individual mapper how exactly certain situations are tagged. For example iD and JOSM have presets for "segregated foot- and cycleway" but they (of course) do not explain what makes a path a "segregated foot- and cycleway", i.e. is a sign required or not?, that's up to the individual mapper. And StreetComplete does the same here, it does not explain whether it must be designated/signed as one or not and this is deliberate, see the posts above. I suspect the actual tagging found in the wild is much more mixed than one would think.

Shoehorning `designated` and `yes` to project the German (unique?) situation of non-mandatory cycle tracks is not particularly meaningful though, as we can't have the same tags meaning something different in different countries. No software will be able to handle this accurately.

<strike>And arguably, it does not even work. Because, if you tag a `segregated` foot- and cycleway that is not mandatory for cyclists because a sign happens to be missing (<a href="https://community.openstreetmap.org/t/radfahrstreifen-in-30-km-h-zonen/5552">for one reason or another</a>) as
```
foot=designated
bicycle=yes
```
then this results in the same access tagging as for footways with cyclists allowed ("Fußweg, Fahrradfahrer frei") although it is a very different situation.
</strike>
_**Edit**: Well, if `segregated` is tagged in that case, it would be possible to distinguish._

If one wants to accurately describe the signed legal situation on the separately mapped cycleway itself, I do not think one can get around mapping the actual `traffic_sign`s that apply, or come up with another tag that describes this concept independent of the specific sign. However, on the latter, the opinion seems to be that whether or not a cycle track is mandatory or not is rather a property of the road than on the cycle track and hence `bicycle=use_sidepath` is already fine for that (which makes sense, if you think about it, IMO)

>> Die Benutzungspflicht lässt sich nur indirekt über das bicycle=use_sidepath an der Straße umsetzen.
>
>Ja, weil nur das logisch einen Sinn macht. Ich kann ja niemanden zwingen einen Radweg zu nutzen der lieber zu Hause bleiben möchte. :yum:

(<a href="https://community.openstreetmap.org/t/fahrradfahrer-frei-von-situationen-unterscheiden-in-denen-radfahrer-implizit-erlaubt-sind/6478/9?u=westnordost
">source</a>)
